test/index.js

```diff
@@ -1,11 +1,32 @@
 // @flow
 import React from 'react';
+import leftPad from 'left-pad'; // eslint-disable-line import/no-extraneous-dependencies, no-unused-vars
+// resolves JS extensions
+import Component from './Component'; // resolves .js
+import ComponentAndroid from './ComponentAndroid'; // resolves .android.js
+import ComponentIos from './ComponentIos'; // resolves .ios.js
+import ComponentNative from './ComponentNative'; // resolves .native.js
+// resolves no TSX? extensions
+import ComponentTS from './ComponentTS'; // eslint-disable-line import/no-unresolved
+import ComponentAndroidTSX from './ComponentAndroidTSX'; // eslint-disable-line import/no-unresolved
```

Comment:
Looking at these tests lines, IMO we should allow .js files to import .ts* files, because current RN typescript template creates such situation by default when index.js is importing App.tsx.

Reply:
IMO that's a bug in TS template, it should import `'App.tsx'` explicitly, instead of just `'App'`. But I've added support to import TS from JS anyway since even Metro supports it